src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -154,318 +142,204 @@ export const defaultMapSingleFIProps: MapSingleFIProps = {
-      polygonFeatureCollection,
-      structures,
-      plansByFocusArea,
-    } = this.props;
-    if (!jurisdiction || !plan) {
-      return <Loading />;
+  React.useEffect(() => {
+    if (props.plan) {
+      /**
+       * Plans present in state e.g when accesing this view from a list of plans or
+       * when the plans are expilcity fetched as in the above if block
+       */
+      fetchData(
+        props.fetchGoalsActionCreator,
+        props.fetchJurisdictionsActionCreator,
+        props.fetchPlansActionCreator,
+        props.fetchStructuresActionCreator,
+        props.fetchTasksActionCreator,
+        props.plan,
+        props.supersetService
+      ).catch((error: Error) => displayError(error));
     }
+    /**
+     * Only re-run effect if props.plan.plan_id changes
+     */
+  }, [props.plan && props.plan.plan_id]);
 
-    /** filter out this plan form plans by focusArea */
-    const otherPlansByFocusArea = plansByFocusArea.filter(localPlan => localPlan.id !== plan.id);
+  React.useEffect(() => {
+    const { setCurrentGoalActionCreator, match } = props;
 
-    const homePage = {
-      label: HOME,
-      url: HOME_URL,
-    };
-    const basePage = {
-      label: FOCUS_INVESTIGATIONS,
-      url: FI_URL,
-    };
-    const secondLastPage = {
-      label: plan.jurisdiction_name,
-      url: `${FI_SINGLE_URL}/${plan.id}`,
-    };
-    const breadCrumbProps: BreadCrumbProps = {
-      currentPage: {
-        label: plan.plan_title,
-        url: `${FI_SINGLE_MAP_URL}/${plan.id}`,
-      },
-      pages: [],
-    };
-    const namePaths =
-      plan.jurisdiction_name_path instanceof Array ? plan.jurisdiction_name_path : [];
-    const pages = namePaths.map((namePath, i) => {
-      // return a page object for each name path
-      return {
-        label: namePath,
-        url: getFilteredFIPlansURL(plan.jurisdiction_path[i], plan.id),
-      };
-    });
-    breadCrumbProps.pages = [homePage, basePage, ...pages, secondLastPage];
-    const statusBadge =
-      plan && plan.plan_status === (PlanStatus.ACTIVE || PlanStatus.DRAFT) ? (
-        <Badge color="warning" pill={true}>
-          {plan.plan_status}
-        </Badge>
-      ) : (
-        <Badge color="success" pill={true}>
-          {plan.plan_status}
-        </Badge>
-      );
-    const markCompleteLink =
-      plan && plan.plan_status === PlanStatus.ACTIVE ? (
-        <Link className="btn btn-primary" to={`${PLAN_COMPLETION_URL}/${plan.id}`} color="primary">
-          {MARK_AS_COMPLETE}
-        </Link>
-      ) : null;
-    /** Array that holds keys of a plan object
-     * Will be used to check plan_effective_period_end or plan_effective_period_start to build the detailview
+    setCurrentGoalActionCreator(match.params.goalId ? match.params.goalId : null);
+    /**
+     * Only re-run effect if props.match.params.goalId changes
      */
-    const planKeysArray: string[] = Object.keys(plan);
+  }, [props.match.params.goalId]);
```

Comment:
This feels like it's skipping the `mapStateToProps` step, would it make sense to reference the `currentGoal` being passed in as props? Would it make more sense to simply pass a `goalId` through the props?

@kelvin-muchiri @moshthepitt @p-netm - what are your thoughts?